### PR TITLE
CIV-8155 Enter flight details screen

### PR DESCRIPTION
**Pull request by hmcts-version1-ignacio**
### Enter flight details screen ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by ShwetaTandel-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/model/FlightDelay.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.reform.civil.model;
+
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/model/FlightDelay.java:2]: I would suggest chaing the name of model to FlightDelayDetail?

File: src/main/java/uk/gov/hmcts/reform/civil/model/FlightDelay.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import java.time.LocalDate;
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class FlightDelay {
+
+    private DynamicList flightDetailsAirlineList;
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/model/FlightDelay.java:18]: If we channge name of pojo to "FlightDelayDetails", we can remove "flightDetails" part from variables names. For example one varialbe will be read "flightDelayDetails.airlineList" than "flightDelay.flightDetailsAirlineList" after changes.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimSpecCallbackHandler.java [hunk 3]
```diff
@@ -525,7 +541,6 @@ private CallbackResponse submitClaim(CallbackParams callbackParams) {
             populateWithPartyIds(dataBuilder);
         }
 
-        List<String> errors = new ArrayList<>();
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimSpecCallbackHandler.java:528]: Why we changed location of this variable?

File: src/main/resources/airline_ePimsID_csv/airline_ePimsID.csv
Comment (ShwetaTandel-hmcts) [inline src/main/resources/airline_ePimsID_csv/airline_ePimsID.csv:1]: Can we move "Other" value to the end of the list?

File: src/main/java/uk/gov/hmcts/reform/civil/model/FlightDelay.java (hunk 1 above)
Comment (hmcts-version1-ignacio) [reply src/main/java/uk/gov/hmcts/reform/civil/model/FlightDelay.java:2]: Yes, agree.

File: src/main/resources/airline_ePimsID_csv/airline_ePimsID.csv
Comment (hmcts-version1-ignacio) [reply src/main/resources/airline_ePimsID_csv/airline_ePimsID.csv:1]: I had it as you say, but @Azam-Hmcts suggested sorting it alphabetically.

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 4]
```diff
@@ -637,6 +637,7 @@ public boolean hasNoOngoingBusinessProcess() {
 
     //SDO-R2
     private YesOrNo isFlightDelayClaim;
+    private FlightDelay flightDelay;
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:640]: Yes, true!

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by Omaira-Melo-Hmcts — DISMISSED** [review]

**Review by ShwetaTandel-hmcts — APPROVED** [review]

**Review by ShwetaTandel-hmcts — APPROVED** [review]

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by Azam-Hmcts — APPROVED** [review]

**Pull request merged**